Assets/LeapMotion/Core/Scripts/Hands/RiggedFinger.cs

```diff
@@ -12,40 +12,95 @@
+    /// </summary>
     [HideInInspector]
     public bool deformPosition = false;
 
+    [HideInInspector]
+    public bool scaleLastFingerBone = false;
+
     public Vector3 modelFingerPointing = Vector3.forward;
     public Vector3 modelPalmFacing = -Vector3.up;
 
     public Quaternion Reorientation() {
       return Quaternion.Inverse(Quaternion.LookRotation(modelFingerPointing, -modelPalmFacing));
     }
 
-    private RiggedHand riggedHand;
+    /// <summary> Backing store for s_standardFingertipLengths, don't touch. </summary>
+    private static float[] s_backingStandardFingertipLengths = null;
+    static RiggedFinger() {
+      // Calculate standard fingertip lengths.
+      s_backingStandardFingertipLengths = new float[5];
+      var testHand = TestHandFactory.MakeTestHand(isLeft: true,
+                           unitType: TestHandFactory.UnitType.UnityUnits);
+      for (int i = 0; i < 5; i++) {
+        var fingertipBone = testHand.Fingers[i].bones[3];
+        s_backingStandardFingertipLengths[i] = fingertipBone.Length;
+      }
+    }
+    /// <summary>
+    /// Fingertip lengths for the standard edit-time hand.
+    /// </summary>
+    private static float[] s_standardFingertipLengths {
+      get {
+        return s_backingStandardFingertipLengths;
+      }
+    }
 
-    /** Updates the bone rotations. */
+    /// <summary>
+    /// Updates model bone positions and rotations based on tracked hand data.
+    /// </summary>
     public override void UpdateFinger() {
       for (int i = 0; i < bones.Length; ++i) {
         if (bones[i] != null) {
           bones[i].rotation = GetBoneRotation(i) * Reorientation();
           if (deformPosition) {
-            bones[i].position = GetJointPosition(i);
+            var boneRootPos = GetJointPosition(i);
+            bones[i].position = boneRootPos;
+
+            if (i == 3 && scaleLastFingerBone) {
+              // Set fingertip base bone scale to match the bone length to the fingertip.
+              // This will only scale correctly if the model was constructed to match
+              // the standard "test" edit-time hand model from the TestHandFactory.
+              var boneTipPos = GetJointPosition(i + 1);
+              var boneVec = boneTipPos - boneRootPos;
+              var boneLen = boneVec.magnitude;
+              var standardLen = s_standardFingertipLengths[(int)this.fingerType];
+              var newScale = bones[i].transform.localScale;
+              var lengthComponentIdx = getLargestComponentIndex(modelFingerPointing);
+              newScale[lengthComponentIdx] = boneLen / standardLen;
+              bones[i].transform.localScale = newScale;
+            }
           }
         }
       }
     }
+
+    private int getLargestComponentIndex(Vector3 pointingVector) {
+      var largestValue = 0f;
+      var largestIdx = 0;
+      for (int i = 0; i < 3; i++) {
+        var testValue = pointingVector[i];
+        if (testValue > largestValue) {
```

Comment:
you need an Abs here, in case the pointing axis points in a negative direction

Reply:
woops good catch, meant to do that but missed it in implementation